### ISSUE TITLE
TextField is not loosing focus

### ISSUE THREAD

**Issue by DarkEye123** [Shopify/polaris-react-archive#7035]
## Issue summary

TextField component of type `number` doesn't loose focus if one of the carets is clicked. Therefore, elements seem to be active even if users are adjusting different form element. 

## Expected behavior
TextField is loosing focus

## Actual behavior
Can be reproduced even on your component library https://polaris.shopify.com/components/text-field -> select type Number. Click on increase/decrease carets and then outside of the element. Observed behavior is that focus is not lost.

For comparison check different type -e .g. email. There it behaves as expected

Tested polaris version -> v10.0.0

**Comment by bkspace** [comment]
This bug was introduced in 9.12.0, so I think the best "safe" version to use until this is fixed is 9.10.0 (9.11.0 has some missing animations I believe)

**Comment by vngauv** [comment]
Looks related to: https://github.com/Shopify/polaris/issues/6385. I'll try taking a look!
